activesupport/test/inspect_test.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+require_relative "abstract_unit"
+require "active_support/inspect"
+
+class InspectTest < ActiveSupport::TestCase
+  class ComplexObject
+    def initialize(id, username, name, interests, password_digest)
+      @id = id
+      @username = username
+      @name = name
+      @interests = interests
+      @password_digest = password_digest
+    end
+
+    include ActiveSupport::Inspect(:@name, :interest_count, :longer_attribute) { [@id, @username.inspect] }
+
+    def interest_count
+      @interests.size
+    end
+
+    def longer_attribute
+      "attribute value"
+    end
+  end
+
+  def setup
+    @object = ComplexObject.new(17, "james", "James", %w(ruby rails), "password")
+  end
+
+  test "#inspect" do
+    assert_equal "#<InspectTest::ComplexObject 17 \"james\" @name=\"James\" interest_count=2 longer_attribute=\"attribute value\">", @object.inspect
+  end
+
+  test "#to_s" do
+    assert_equal "#<InspectTest::ComplexObject 17 \"james\">", @object.to_s
+  end
+
+  test "#pretty_print" do
+    output = +""
+    PP.pp(@object, output, 20)
+    assert_equal <<~OUT, output
+    #<InspectTest::ComplexObject
+     17 "james"
+     @name="James"
+     interest_count=2
+     longer_attribute=
+      "attribute value">
+    OUT
+  end
+
+  class InheritanceParent
+    def inspect
+      "custom inspect"
+    end
+
+    def to_s
+      "custom to_s"
+    end
+  end
+
+  class ChildWithInspect < InheritanceParent
+    include ActiveSupport::Inspect()
```

Comment:
Could we rename the `Inspect` class to something else, and then have `ActiveSupport::Inspect` be a default instance?

It would both get rid of the surprising `()` and allow to share that one module more, hence reduce mem usage a bit.